nativeio_async_wrapper.js

```diff
@@ -180,6 +192,45 @@ mergeInto(LibraryManager.library, {
     })
   },
 
+  NativeIO_SetLength__deps: ['$NativeIOWrapper','$NativeIOUtils', '$Asyncify'],
+  NativeIO_SetLength: function(fileDescriptor, {{{ defineI64Param('length') }}}) {
+    return Asyncify.handleSleep(function(wakeUp) {
+      {{{ receiveI64ParamAsI32s('length') }}}
+      try {
+        var length = NativeIOUtils.combineI64Params(length_low, length_high);
+      } catch (error) {
+        console.log(
+          'NativeIO error receiving length to truncate file with file descriptor number',
+           fileDescriptor + ':', error);
+        wakeUp(-{{{cDefine('EINVAL')}}});
+      }
+
+      NativeIOWrapper.setLength(fileDescriptor, length).then(
+        () => {wakeUp(0)},
+        (error) => {
+          console.log(
+            'NativeIO error while setting length of file with file descriptor number',
+            fileDescriptor, 'to', offset + ':', error);
+          wakeUp(-{{{cDefine('EINVAL')}}});
+        })
+    })
+  },
+
+  NativeIO_GetLength__deps: ['$NativeIOWrapper','$NativeIOUtils', '$Asyncify'],
+  NativeIO_GetLength: function(fileDescriptor) {
+    return Asyncify.handleSleep(function(wakeUp) {
+      NativeIOWrapper.getLength(fileDescriptor).then(
+        (length) => {
+          wakeUp(length)},
```

Comment:
Nit: line break after } ?

Reply:
I moved everything into one line for consistency.